inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/reduce/reduce_kernel_b_fs_yx_fsv16.cpp

```diff
@@ -0,0 +1,140 @@
+/*
+// Copyright (c) 2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+#include "reduce_kernel_b_fs_yx_fsv16.h"
+#include "kernel_selector_utils.h"
+#include <vector>
+#include <string>
+#include "common_tools.h"
+
+namespace kernel_selector {
+
+static const size_t SIMD = 16;
+using NDims = std::vector<kernel_selector::Tensor::Dim>;
+
+size_t calc_read_offset(const reduce_params& params) {
+    auto read_offset = 1;
+    if (params.inputs[0].GetDType() == Datatype::F32 || params.inputs[0].GetDType() == Datatype::INT32)
+        read_offset = 4;
+    else if (params.inputs[0].GetDType() == Datatype::F16)
+        read_offset = 8;
+    else if (params.inputs[0].GetDType() == Datatype::INT8)
```

Comment:
What about `UINT8`? In general wouldn't it be safer to select based on element size than on explicit types?